keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -1199,6 +1200,41 @@ class SqlArtifactRepository(
     }
   }
 
```

Comment:
It's probably worth adding a comment here that says "This will fetch the most recent artifact by the 'deployed_at" field"

Also, because you're sorting by deployed at, it won't work as you expect for artifacts that are APPROVED or SKIPPED or DEPLOYING. Do you think it's worth making that obvious by adding a:

```kotlin
require(promotionStatus in listOf(CURRENT, PREVIOUS)) { "Invalid promotion status used to query" }
```
(the require function is a kotlin thing that makes argument checking easier: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/require.html)

Reply:
@emjburns for `DEPLOYING`, I'm comparing against `CURRENT`. (`SKIPPED` and `APPROVED` are out of scope for this PR, for now, but I think it would make sense to compare them against the current as well). Do you still think it can cause issues?

Reply:
added `require`.